code/game/objects/items/RCD.dm

```diff
@@ -636,14 +625,24 @@ GLOBAL_VAR_INIT(icon_holographic_window, init_holographic_window())
 				balloon_alert(user, "something is on the tile!")
 				qdel(rcd_effect)
 				return FALSE
+	//beam animation for arcd
+	var/beam
+	if(ranged)
+		beam = user.Beam(A,icon_state="rped_upgrade", time = delay)
 	if(!do_after(user, delay, target = A))
 		qdel(rcd_effect)
+		if(beam)
```

Comment:
```suggestion
		if(!isnull(beam))
```

faster, and more explicit as to why we're doing this check in the first place. replicate this behavior to every other instance of this pattern.